apps/dapp-console-api/src/routes/apps/AppsRoute.ts

```diff
@@ -0,0 +1,54 @@
+import { generateListResponse, zodListRequest, zodNameCursor } from '@/api'
+import { isPrivyAuthed } from '@/middleware'
+import { getActiveAppsForEntityByCursor } from '@/models'
+import { metrics } from '@/monitoring/metrics'
+import { Trpc } from '@/Trpc'
+
+import { DEFAULT_PAGE_LIMIT } from '../constants'
+import { Route } from '../Route'
+
+export class AppsRoute extends Route {
+  public readonly name = 'apps' as const
+
+  public readonly listApps = 'listApps' as const
+  /**
+   * Returns a list of apps associated with the entity.
+   */
+  public readonly listAppsController = this.trpc.procedure
+    .use(isPrivyAuthed(this.trpc))
+    .input(zodListRequest(zodNameCursor))
+    .query(async ({ ctx, input }) => {
+      try {
+        const { user } = ctx.session
+        const limit = input.limit ?? DEFAULT_PAGE_LIMIT
+
+        if (!user) {
+          throw Trpc.handleStatus(401, 'user not authenticated')
+        }
+
+        const activeApps = await getActiveAppsForEntityByCursor({
```

Comment:
for now, I think that this endpoint should only load the apps and we should have a separate endpoint for loading the contracts associated with an app. Other option would be to load the contracts alongside the apps and have this endpoint return a list of apps along with a list of contracts within those apps. I like having two separate endpoints with each one returning its own cursor since that will simplify things but downside is it requires an extra call from the client. wdyt?

Reply:
ok, yeah, that definitely makes sense. updated to return contracts alongside each app.